rfcs/BFDv2/Patient RFC.md

```diff
@@ -0,0 +1,250 @@
+# RFC Proposal
+
+[RFC Proposal]: #rfc-proposal
+
+* RFC Proposal ID: `0000-patient-fhir-r4-carin_bfdv2`
+* Start Date: 08-03-2020
+* RFC PR: [BFD-264-v2_Patient_RFC #313](https://github.com/CMSgov/beneficiary-fhir-data/pull/313)
+* JIRA Ticket(s): [BFD-264](https://jira.cms.gov/browse/BFD-264)
+
+
+This RFC proposes that our team migrate from BFD v1 STU3 Patient FHIR resource to the first normative FHIR release, R4 and implement CARIN Patient profile.
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Proposed Solution](#proposed-solution)
+	* [Detailed Design](#detailed-design)
+	* [Current State](#current-state)
+	* [Future State](#future-state)
+	* [Differential View](#differential-view)
+	* [Unresolved Questions](#unresolved-questions)
+	* [Drawbacks](#proposed-solution-drawbacks)
+	* [Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+
+## Motivation
+[Motivation]: #motivation
+
+This change will make BFD's Patient FHIR resource increase the system's conformance with FHIR R4 and industry alliance like CARIN.
+This will help BFD move its platform to use the first normative release of FHIR R4. This will also make integration and 
+interoperability with all our peer partners seamless by enabling to provide FHIR R4 validated and conformant Patient resources
+(Beneficiaries demographic data).
+
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+
+BFD proposes to provide a new v2 endpoint based on Patient FHRI-R4 and CARIN BB Profile. 
+
+Sample indicative Patient v2 endpoint:  
+https://localhost:1337/v2/fhir/Patient?_id=-19990000000001&_format=json
+
+
+Patient FHIR R4 (US-Core, Normative release):  
+http://hl7.org/fhir/R4/patient.html
+
+
+CARIN Patient Profile:  
+http://build.fhir.org/ig/HL7/carin-bb/StructureDefinition-CARIN-BB-Patient.html
+
+### Detailed Design
+
+[Detailed Design]: #detailed-design
+
+**HAPI Version to be used:** HAPI FHIR 4.1.0
+
+**HAPI 4.1.0 Patient Data Model class to be used:** org.hl7.fhir.r4.model.Patient
+
+**POC Code snippet:** 
+
+	// Slicing for Patient Identifier
+      Patient patientSlicing = new Patient();
+
+      patientSlicing.addIdentifier().setValue("-20140000010000")
+          .setSystem("https://bluebutton.cms.gov/resources/variables/bene_id").getType().addCoding()
+          .setCode(
+          "PI")
+          .setSystem("http://terminology.hl7.org/CodeSystem/v2-0203")
```

Comment:
Are you sure this is the correct system? It seems to be for `Identifier.type`, instead of `Coding.system`. Apologies if I'm reading it wrong.